inference-engine/src/preprocessing/ie_preprocess_gapi_kernels.cpp

```diff
@@ -774,15 +818,225 @@ static void chanToPlaneRow(const uint8_t* in, int chan, int chs, uint8_t* out, i
 //        }
 //    };
 
+namespace {
+
+using chan_to_plane_supported_types = typelist<uint8_t, float>;
+
+template<typename T>
+void chanToPlaneRowImpl(scalar_tag, const T* in, int chan, int chs, T* out, int length) {
+    for (int x = 0; x < length; x++) {
+        out[x] = in[x*chs + chan];
+    }
+}
+
+template<typename isa_tag_t>
+struct typed_chan_to_plane_row {
+    using p_f = void (*)(const uint8_t* in, int chan, int chs, uint8_t* out, int length);
+
+    template <typename type>
+    p_f operator()(type_to_type<type> ) {
```

Comment:
i'm sorry  for my confusion, but is there a pointer on function which points on local-scope lambda?
i mean, does it work on dangling pointer here?

may be `using pf = std::function<void()> ` is more safe here?

Reply:
Ok.I try to apply this comment in next PRs-inheritors with refactoring for Resize kernels.